File: src/main/java/com/example/eyeofsauron/repository/RouteRepository.java
package com.example.eyeofsauron.repository;

import com.example.eyeofsauron.entity.Route;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface RouteRepository extends JpaRepository<Route, Long>, JpaSpecificationExecutor<Route> {
    List<Route> getBySecuredFacilityId(Long id);
}

File: src/main/kotlin/com/example/eyeofsauron/repository/RouteRepository.kt
package com.example.eyeofsauron.repository

import com.example.eyeofsauron.entity.Route
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository

@Repository
interface RouteRepository : JpaRepository<Route, Long>, JpaSpecificationExecutor<Route> {
    fun findBySecuredFacilityId(id: Long): List<Route>
}